src/Main.elm

```diff
@@ -3,77 +3,432 @@ module Main exposing (main)
 import Browser
 import Data
 import Html exposing (Html)
-import Html.Attributes as Attrs
+import Json.Decode
+import Element as E exposing (Element)
+import Element.Font as F
+import Ui.Theme as T
+import Ui
+import Validate
 
 
+{-
 
----- MODEL ----
+TODO
+
+- Debounce so errors show up after finished typing.
+- Arrange validations errors such that submit button doesn't jump.
+- Highlight invalid input fields and add Aria.invalid tags.
+- Maybe refocus on invalid field when attempting to submit instead of
+  disabling submit button.
```

Comment:
There is a risk that it could be a bit confusing for the user, but this is an interesting idea!